app/app.go

```diff
@@ -465,48 +468,139 @@ func NewSifApp(
 	// During begin block slashing happens after distr.BeginBlocker so that
 	// there is nothing left over in the validator fee pool, so as to keep the
 	// CanWithdrawInvariant invariant.
-	app.mm.SetOrderBeginBlockers(
-		upgradetypes.ModuleName,
-		capabilitytypes.ModuleName,
-		minttypes.ModuleName,
-		distrtypes.ModuleName,
-		slashingtypes.ModuleName,
-		evidencetypes.ModuleName,
-		stakingtypes.ModuleName,
-		ibchost.ModuleName,
-		dispensation.ModuleName,
-		clptypes.ModuleName,
-	)
-	app.mm.SetOrderEndBlockers(
-		crisistypes.ModuleName,
-		govtypes.ModuleName,
-		stakingtypes.ModuleName,
-		feegrant.ModuleName,
-		clptypes.ModuleName,
-	)
+	if featureToggleSdk045 {
+		// NOTE: staking module is required if HistoricalEntries param > 0
+		// NOTE: capability module's beginblocker must come before any modules using capabilities (e.g. IBC)
+		app.mm.SetOrderBeginBlockers(
+			upgradetypes.ModuleName,
+			capabilitytypes.ModuleName,
+			minttypes.ModuleName,
+			distrtypes.ModuleName,
+			slashingtypes.ModuleName,
+			evidencetypes.ModuleName,
+			stakingtypes.ModuleName,
+			authtypes.ModuleName,
+			banktypes.ModuleName,
+			govtypes.ModuleName,
+			crisistypes.ModuleName,
+			genutiltypes.ModuleName,
+			feegrant.ModuleName,
+			paramstypes.ModuleName,
+			vestingtypes.ModuleName,
+			ibchost.ModuleName,
+			disptypes.ModuleName,
+			transferModule.Name(),
+			vestingtypes.ModuleName,
+			clptypes.ModuleName,
+			ethbridgetypes.ModuleName,
+			tokenregistrytypes.ModuleName,
+			oracletypes.ModuleName,
+			dispensation.ModuleName,
+		)
+	} else {
+		app.mm.SetOrderBeginBlockers(
+			upgradetypes.ModuleName,
+			capabilitytypes.ModuleName,
+			minttypes.ModuleName,
+			distrtypes.ModuleName,
+			slashingtypes.ModuleName,
+			evidencetypes.ModuleName,
+			stakingtypes.ModuleName,
+			ibchost.ModuleName,
+			dispensation.ModuleName,
+			clptypes.ModuleName,
+		)
+	}
+	if featureToggleSdk045 {
```

Comment:
It might a good idea to have only 1  if/else block and set all functions in that